client/my-sites/checkout/checkout-thank-you/header.jsx

```diff
@@ -618,11 +633,22 @@ export class CheckoutThankYouHeader extends PureComponent {
 }
 
 export default connect(
-	( state, ownProps ) => ( {
-		upgradeIntent: ownProps.upgradeIntent || getCheckoutUpgradeIntent( state ),
-		isAtomic: isAtomicSite( state, ownProps.selectedSite?.ID ),
-		siteAdminUrl: getSiteAdminUrl( state, ownProps.selectedSite?.ID ),
-	} ),
+	( state, ownProps ) => {
+		let isPendingDomainUpdateForTitan = false;
+		if (
+			ownProps.primaryPurchase &&
+			ownProps.selectedSite?.ID &&
+			isTitanMail( ownProps.primaryPurchase )
+		) {
+			isPendingDomainUpdateForTitan = isRequestingSiteDomains( state, ownProps.selectedSite.ID );
```

Comment:
Isn't it something [already handled](https://github.com/Automattic/wp-calypso/blob/43d3beb3962b37f72ea2f2f0f78c6e480c0b758e/client/components/data/query-site-domains/index.jsx#L15) by the `QuerySiteDomains` component?

Reply:
It is, but I am surfacing this in the component so that we know when we're waiting for a request triggered by this component -- I don't want this flag to be checked for non-Titan purchases, especially because we don't want to trigger the request for non-Titan purchases.